src/libraries/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.cs

```diff
@@ -685,8 +685,15 @@ internal void FinishWrapperConnectSuccess(Socket? connectSocket, int bytesTransf
             if (SocketsTelemetry.Log.IsEnabled()) LogBytesTransferEvents(connectSocket?.SocketType, SocketAsyncOperation.Connect, bytesTransferred);
 
             // Complete the operation and raise the event.
-            ExecutionContext? context = _context; // store context before it's cleared as part of completing the operation
             Complete();
+        }
+
+        internal void FinishWrapperConnectAsyncSuccess(Socket? connectSocket, int bytesTransferred, SocketFlags flags)
+        {
+            ExecutionContext? context = _context; // store context before it's cleared as part of completing the operation
+
+            FinishWrapperConnectSyncSuccess(connectSocket, bytesTransferred, flags);
+
```

Comment:
Doesn't this change mean we're no longer calling OnCompletedInternal for sync completions?

Reply:
Yes, we're no longer calling OnCompletedInternal for sync completions. That's the source of the bug -- we were both returning pending=false and calling the OnCompleted callback. Now we just return pending=false.

OnCompletedInternal doesn't log for the multi connect case anyway, there's an explicit check for it. I believe how it works is that it logs per connection attempt.

@MihaZupan Does that all sound right?

Reply:
Yes, for the Socket's connect telemetry we're only logging connection attempts, not the parent MultiConnect operation as a whole